src/lib.rs

```diff
@@ -22,152 +17,76 @@ pub struct Component {
-        match self {
-            CommandInstancePre::Component(pre) => pre
-                .instantiate_async(store)
-                .await
-                .map(CommandInstance::Component),
-            CommandInstancePre::Module(pre) => pre
-                .instantiate_async(store)
-                .await
-                .map(CommandInstance::Module),
-        }
+    async fn run(self, trigger_app: spin_trigger::TriggerApp<Self>) -> anyhow::Result<()> {
+        Self::handle(
+            self.components
+                .get(0)
+                .map(|component| component.to_owned())
+                .context("Failed to get the component for the command trigger")?,
+            trigger_app.into(),
+            self.config.clone(),
+        )
+        .await
     }
 }
 
 impl CommandTrigger {
-    pub async fn handle(&self, args: CliArgs) -> Result<()> {
-        let component = &self.components[0];
-        let mut store_builder = self
-            .engine
-            .store_builder(&component.id, spin_core::WasiVersion::Preview2)?;
-
-        args.apply_args_to_store(&component.id, &mut store_builder)?;
-
-        let (instance, mut store) = self
-            .engine
-            .prepare_instance_with_store(&component.id, store_builder)
-            .await?;
-        match instance {
-            CommandInstance::Component(instance) => {
-                let handler = Command::new(&mut store, &instance)
-                    .context("Wasi preview 2 components need to target the wasi:cli world")?;
-                let _ = handler.wasi_cli_run().call_run(store).await?;
-            }
-            CommandInstance::Module(_) => {
-                // Toss the commandInstance we have and create a new one as the
-                // associated store will be a preview2 store
-                let mut store_builder = self
-                    .engine
-                    .store_builder(&component.id, spin_core::WasiVersion::Preview1)?;
-
-                args.apply_args_to_store(&component.id, &mut store_builder)?;
-
-                let (instance, mut store) = self
-                    .engine
-                    .prepare_instance_with_store(&component.id, store_builder)
-                    .await?;
-                let CommandInstance::Module(instance) = instance else {
-                    unreachable!();
-                };
-
-                let start = instance
-                    .get_func(&mut store, "_start")
-                    .context("Expected component to export _start function")?;
-
-                start.call_async(&mut store, &[], &mut []).await?;
-            }
-        }
+    pub async fn handle(
+        component: Component,
+        trigger_app: Arc<TriggerApp<Self>>,
+        args: CliArgs,
+    ) -> Result<()> {
+        let mut instance_builder = trigger_app.prepare(&component.id)?;
+        let t = instance_builder.factor_builders().wasi();
+        t.args(args.guest_args);
```

Comment:
The prev version included the component ID as the first arg here. Is that an intentional change?

Reply:
Nope, just glossed over it, I will update.

Reply:
fixed